pkg/bootstrap/platform/gcp_test.go

```diff
@@ -183,13 +209,77 @@ func TestGCPMetadata(t *testing.T) {
+			e := NewGCP()
+			got := e.Metadata()
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("gcpEnv.Metadata() => '%v'; want '%v'", got, tt.want)
+			}
+			os.Clearenv()
+			envOnce, envPid, envNpid, envCluster, envLocation = sync.Once{}, "", "", "", ""
+		})
+	}
+}
+
+func TestMetadataCache(t *testing.T) {
+	tests := []struct {
+		name                string
+		shouldFill          shouldFillFn
+		projectIDFn         metadataFn
+		numericProjectIDFn  metadataFn
+		locationFn          metadataFn
+		clusterNameFn       metadataFn
+		instanceNameFn      metadataFn
+		instanceIDFn        metadataFn
+		instanceTemplateFn  metadataFn
+		instanceCreatedByFn metadataFn
+		env                 map[string]string
+		want                map[string]string
+	}{
+		{
+			"should cache",
+			func() bool { return true },
+			func() (string, error) { return "pid", nil },
+			func() (string, error) { return "npid", nil },
+			func() (string, error) { return "location", nil },
+			func() (string, error) { return "cluster", nil },
+			func() (string, error) { return "instanceName", nil },
+			func() (string, error) { return "instance", nil },
+			func() (string, error) { return "instanceTemplate", nil },
+			func() (string, error) { return "createdBy", nil },
+			map[string]string{},
+			map[string]string{GCPProject: "pid", GCPProjectNumber: "npid", GCPLocation: "location", GCPCluster: "cluster", GCEInstance: "instanceName",
+				GCEInstanceID: "instance", GCEInstanceTemplate: "instanceTemplate", GCEInstanceCreatedBy: "createdBy"},
+		}, {
+			"should  ignore",
+			func() bool { return true },
+			func() (string, error) { return "newPid", nil },
+			func() (string, error) { return "newNpid", nil },
+			func() (string, error) { return "newLocation", nil },
+			func() (string, error) { return "newCluster", nil },
+			func() (string, error) { return "newInstanceName", nil },
+			func() (string, error) { return "newInstance", nil },
+			func() (string, error) { return "newInstanceTemplate", nil },
+			func() (string, error) { return "newCreatedBy", nil },
+			map[string]string{},
+			map[string]string{GCPProject: "pid", GCPProjectNumber: "npid", GCPLocation: "location", GCPCluster: "cluster", GCEInstance: "instanceName",
+				GCEInstanceID: "instance", GCEInstanceTemplate: "instanceTemplate", GCEInstanceCreatedBy: "createdBy"},
+		},
+	}
+	gcpEnv := NewGCP()
+	for idx, tt := range tests {
+		t.Run(fmt.Sprintf("[%d] %s", idx, tt.name), func(t *testing.T) {
+			shouldFillMetadata, projectIDFn, numericProjectIDFn, clusterLocationFn, clusterNameFn, instanceNameFn, instanceIDFn, instanceTemplateFn, createdByFn =
+				tt.shouldFill, tt.projectIDFn, tt.numericProjectIDFn, tt.locationFn, tt.clusterNameFn, tt.instanceNameFn, tt.instanceIDFn,
+				tt.instanceTemplateFn, tt.instanceCreatedByFn
+			got := gcpEnv.Metadata()
 			if !reflect.DeepEqual(got, tt.want) {
 				t.Errorf("gcpEnv.Metadata() => '%v'; want '%v'", got, tt.want)
 			}
+			os.Clearenv()
```

Comment:
The current tests don't set any env variables, but in case that future tests are added with env variables I thought it would be better to keep it there since env vars persist across tests.